pallets/did/src/errors.rs

```diff
@@ -22,59 +22,93 @@ use crate::did_details::DidVerificationKeyRelationship;
 
 /// All the errors that can be generated when validating a DID operation.
 #[derive(Debug, Eq, PartialEq, TypeInfo)]
-pub enum DidError {
-	/// See [StorageError].
-	StorageError(StorageError),
-	/// See [SignatureError].
-	SignatureError(SignatureError),
-	/// See [InputError].
-	InputError(InputError),
+pub enum Error {
+	/// See [Storage].
+	Storage(Storage),
+	/// See [Signature].
+	Signature(Signature),
+	/// See [Input].
+	Input(Input),
 	/// An error that is not supposed to take place, yet it happened.
-	InternalError,
+	Internal,
 }
 
-impl From<StorageError> for DidError {
-	fn from(err: StorageError) -> Self {
-		DidError::StorageError(err)
+impl From<Storage> for Error {
+	fn from(err: Storage) -> Self {
+		Error::Storage(err)
 	}
 }
 
-impl From<InputError> for DidError {
-	fn from(err: InputError) -> Self {
-		DidError::InputError(err)
+impl From<Input> for Error {
+	fn from(err: Input) -> Self {
+		Error::Input(err)
 	}
 }
 
 /// Error involving the pallet's storage.
 #[derive(Debug, Eq, PartialEq, TypeInfo)]
-pub enum StorageError {
+pub enum Storage {
 	/// The DID being created is already present on chain.
-	DidAlreadyPresent,
+	AlreadyExists,
 	/// The expected DID cannot be found on chain.
-	DidNotPresent,
-	/// The given DID does not contain the right key to verify the signature
-	/// of a DID operation.
-	DidKeyNotPresent(DidVerificationKeyRelationship),
-	/// At least one key referenced is not stored under the given DID.
-	KeyNotPresent,
+	NotFound(NotFoundKind),
 	/// The maximum number of public keys for this DID key identifier has
 	/// been reached.
-	MaxPublicKeysPerDidExceeded,
+	MaxPublicKeysExceeded,
 	/// The maximum number of key agreements has been reached for the DID
 	/// subject.
 	MaxTotalKeyAgreementKeysExceeded,
 	/// The DID has already been previously deleted.
-	DidAlreadyDeleted,
+	AlreadyDeleted,
+}
+
+/// Error involving the pallet's storage.
+#[derive(Debug, Eq, PartialEq, TypeInfo)]
+pub enum NotFoundKind {
+	/// The expected DID cannot be found on chain.
+	Did,
+	/// At least one key referenced is not stored under the given DID.
+	Key(KeyType),
+}
+
+/// Enum describing the different did key types.
+#[derive(Debug, Eq, PartialEq, TypeInfo)]
+pub enum KeyType {
```

Comment:
Even tho this introduces another nesting level i would maybe reuse the `DidVerificationKeyRelationship` here.

```
pub enum KeyType {
 Verification(DidVerificationKeyRelationship),
 KeyAgreement,
}
```

You can than implement `From<DidVerificationKeyRelationship> for KeyType` or maybe even derive that and `From<KeyType> for NotFoundKind`

When creating an error gets too deeply nested you can also add a helper function to `StorageError`.